presto-testing/src/main/java/io/prestosql/testing/AbstractTestQueries.java

```diff
@@ -1807,15 +1807,15 @@ public void testAccessControl()
         assertAccessAllowed("SHOW STATS FOR lineitem", privilege("orders", SELECT_COLUMN));
         assertAccessAllowed("SHOW STATS FOR (SELECT * FROM lineitem)");
         assertAccessAllowed("SHOW STATS FOR (SELECT * FROM lineitem)", privilege("orders", SELECT_COLUMN));
-        assertAccessDenied("SHOW STATS FOR (SELECT * FROM nation)", "Cannot show stats for columns \\[nationkey, regionkey, name, comment\\] in table or view .*.nation", privilege("nation", SELECT_COLUMN));
-        assertAccessDenied("SHOW STATS FOR (SELECT nationkey FROM nation)", "Cannot show stats for columns \\[nationkey\\] in table or view .*.nation", privilege("nation", SELECT_COLUMN));
-        assertAccessDenied("SHOW STATS FOR (SELECT nationkey FROM nation)", "Cannot show stats for columns \\[nationkey\\] in table or view .*.nation", privilege("nation.nationkey", SELECT_COLUMN));
-        assertAccessDenied("SHOW STATS FOR (SELECT *, nationkey FROM nation)", "Cannot show stats for columns \\[nationkey, regionkey, name, comment\\] in table or view .*.nation", privilege("nation.nationkey", SELECT_COLUMN));
-        assertAccessDenied("SHOW STATS FOR (SELECT *, * FROM nation)", "Cannot show stats for columns \\[nationkey, regionkey, name, comment\\] in table or view .*.nation", privilege("nation.nationkey", SELECT_COLUMN));
-        assertAccessDenied("SHOW STATS FOR (SELECT linenumber, orderkey FROM lineitem)", "Cannot show stats for columns \\[linenumber, orderkey\\] in table or view .*.lineitem.*", privilege("lineitem", SELECT_COLUMN));
-        assertAccessDenied("SHOW STATS FOR (SELECT linenumber, orderkey, quantity FROM lineitem)", "Cannot show stats for columns \\[linenumber, orderkey, quantity\\] in table or view .*.lineitem.*", privilege("lineitem.linenumber", SELECT_COLUMN), privilege("lineitem.orderkey", SELECT_COLUMN));
-        assertAccessDenied("SHOW STATS FOR (SELECT nationkey FROM nation)", "Cannot show stats for columns \\[nationkey\\] in table or view .*.nation.*", privilege("nation", SELECT_COLUMN));
-        assertAccessDenied("SHOW STATS FOR (SELECT * FROM nation)", "Cannot show stats for columns \\[nationkey, regionkey, name, comment\\] in table or view .*.nation.*", privilege("nation", SELECT_COLUMN));
+        assertAccessDenied("SHOW STATS FOR (SELECT * FROM nation)", "Cannot select from columns \\[nationkey, regionkey, name, comment\\] in table or view .*.nation", privilege("nation", SELECT_COLUMN));
```

Comment:
Original error message was more accurate

Reply:
Yes, the original was a wrapping one, that goes to the same bucket as: https://github.com/prestosql/presto/pull/5379#discussion_r514403546

Should I extract that change to separate commit?

Reply:
Naa - if that is the way to go I do not care about extracting as separate commit.

Reply:
i raised similar problem over here https://github.com/prestosql/presto/pull/3109#discussion_r528601536.

I am not convinced by the linked comment.
Especially for case `SHOW STATS FOR table_name` it seems a bit weird to throw "Cannot select from columns ...".

Reply:
This is like `CALL create_empty_partitions(...)` that throws "Cannot insert into...".

It is not about being not allowed to call `CALL create_empty_partitions()` or `SHOW STATS FOR x` but not being able to create new partition (and so doing an `insert`) or not being able to see table data (and so doing a `select`). So this is more detailed. Also it is more actionable, as now you know that you need to `GRANT INSERT` or `GRANT SELECT`, while there is no `GRANT CALL` or `GRANT SHOW STATS`.

Surely this is one way of thinking. I was thinking differently here: https://github.com/prestosql/presto/pull/5379#discussion_r511917063. Since after few iterations https://github.com/prestosql/presto/pull/5379 changed, I decided to change this pull request into the same direction to be coherent.